packages/pigeon/lib/swift_generator.dart

```diff
@@ -0,0 +1,642 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:pigeon/functional.dart';
+
+import 'ast.dart';
+import 'generator_tools.dart';
+
+/// Options that control how Swift code will be generated.
+class SwiftOptions {
+  /// Creates a [SwiftOptions] object
+  const SwiftOptions({
+    this.prefix,
+    this.copyrightHeader,
+  });
+
+  /// Prefix that will be appended before all generated classes and protocols.
+  final String? prefix;
```

Comment:
In fact, Swift doesn't.
The only reason for me add that was for testing. Unlike other languages (ObjC, Java, Kotlin) Swift does't include/import classes by file.
And, since the sample pigeons has classes with same names, happens a conflict if there not be a prefix.
Unfortunately, this was the only workaround I found for this at the moment.

Reply:
I don't follow; what is the collision here that can't be resolved with module names (which is how collisions are supposed to be resolved in Swift, thus the lack of prefixing)?

Reply:
I agree.
But in this case, all classes are in the same module.
The script generates the files `platform_tests/ios_swift_unit_tests/ios/Runner/*.gen.swift` to test.
But, as `pigeons/flutter_unittests.dart` as `pigeons/host2flutter.dart` has the class `SearchReply` the Swift compile fail with `Swift Compiler Error (Xcode): Invalid redeclaration of 'SearchReply'`
There are other classes with this conflict.

A workaround for this would be rename this pigeons classes in order to make them different among theirs.
Can I proceed with that?

Reply:
If the problem is the unit tests, we should absolutely change the unit tests to work rather than add functionality to the public API that should not exist.

Reply:
Done. I've removed the prefix feature for Swift generator and change the names of pigeons data classes and apis.